src/onnx/parse_rnn.cpp

```diff
@@ -33,6 +33,29 @@ namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 namespace onnx {
 
+void rnn_transpose_inputs(onnx_parser::node_info& info, std::vector<instruction_ref>& args)
+{
+    std::vector<int64_t> perm{1, 0, 2};
+    args[0] = info.add_instruction(make_op("transpose", {{"permutation", perm}}), args[0]);
+
+    if(args.size() == 6 and not args[5]->is_undefined())
+    {
+        args[5] = info.add_instruction(make_op("transpose", {{"permutation", perm}}), args[5]);
```

Comment:
Where is this args[5] consumed, please?

Reply:
As in the https://onnx.ai/onnx/operators/onnx__RNN.html, the args[5] represent the optional argument initial_h, if the argument is not undefined it is used in calculations present in src/rewrite_rnn.cpp (rewrite_rnn::apply_vanilla_rnn)

Reply:
Thanks.